### PR TITLE
Fixes #15196: make the  remote run agent parameters configurable

### PR DESCRIPTION
**Pull request by ArmandSy**
https://issues.rudder.io/issues/15196

### CODE REVIEW

File: relay/sources/relayd/src/configuration.rs [hunk 1]
```diff
@@ -167,6 +168,13 @@ impl OutputSelect for InventoryOutputSelect {
     }
 }
 
+#[derive(Deserialize, Debug, PartialEq, Eq, Clone)]
+
+pub struct RemoteRun
+{
+    pub command : PathBuf,
```
Comment (amousset) [inline relay/sources/relayd/src/configuration.rs]: No space before :, but `cargo fmt` should take care of that

File: relay/sources/relayd/src/remote_run.rs [hunk 2]
```diff
@@ -175,14 +176,13 @@ impl RunParameters {
...
         let program = if test_mode {
-            "echo"
+            PathBuf::from("echo")
         } else {
-            // FIXME make it configurable
-            "/opt/rudder/bin/rudder"
-        }
-        .to_string();
+            let jobconfig2 = &job_config.clone();
+            PathBuf::from(jobconfig2.clone().cfg.remote_run.command.to_str().unwrap())
```
Comment (amousset) [inline relay/sources/relayd/src/remote_run.rs]: `job_config.clone().cfg.remote_run.command.clone()` should be enough

File: relay/sources/relayd/src/remote_run.rs [hunk 3]
```diff
@@ -38,6 +38,7 @@ use std::{
     process::{Command, Stdio},
     str::FromStr,
     sync::Arc,
+    path::{Path, PathBuf},
```
Comment (amousset) [inline relay/sources/relayd/src/remote_run.rs]: `Path` is not used and produces a warning, we should not have warnings in merged code.

File: relay/sources/relayd/src/remote_run.rs (hunk 2 above)
Comment (amousset) [reply relay/sources/relayd/src/remote_run.rs]: We have to clone because a reference is not enough as we have to return an owned `PathBuf`

**Comment by ArmandSy** [comment]
Commit modified

**Pull request merged**